android/PhysicalWeb/app/src/test/java/org/physical_web/physicalweb/UtilsTest.java

```diff
@@ -0,0 +1,634 @@
+   *
+   * Procedure:
+   *  1. Verify string with null values and 0
+   *  2. Verify string with empty values and the minimum integer value
+   *  3. Verify string with non empty values and the maximum integer value
+   */
+  @Test
+  public void testFormatEndpointForSharedPreferences() throws Exception {
+    String url = null;
+    int version = 0;
+    String key = null;
+    Assert.assertArrayEquals("Null values", new String[]{"", Integer.toString(version)},
+        Utils.formatEndpointForSharedPrefernces(url, version, key).split("\0"));
+    url = "";
+    version = Integer.MIN_VALUE;
+    key = "";
+    Assert.assertArrayEquals("Empty Strings and min", new String[]{url, Integer.toString(version)},
+        Utils.formatEndpointForSharedPrefernces(url, version, key).split("\0"));
+    url = "url";
+    version = Integer.MAX_VALUE;
+    key = SAMPLE_API_KEY;
+    Assert.assertArrayEquals("Non empty strings and max",
+        new String[]{url, Integer.toString(version), key},
+        Utils.formatEndpointForSharedPrefernces(url, version, key).split("\0"));
+  }
+
+  /*
+   * Tests getDefaultPwsEndpointPreferenceString. Also exercises isGoogleApiKeyAvailable and
+   * formatEndpointForSharedPreferences.
+   *
+   * Procedure:
+   *  1. Configure there to not be an api key
+   *  2. Verify default is the prod pws
+   *  3. Configure there to be an api key
+   *  4. Verify default is the Google pws
+   */
+  @Test
+  public void testGetDefaultPwsEndpointPreferenceString() throws Exception {
+    int resId = 1;
+    String apiKey = SAMPLE_API_KEY;
+    String packageName = SAMPLE_PACKAGE_NAME;
+    Resources mockResources = Mockito.mock(Resources.class);
+    Mockito.when(mockResources.getIdentifier(GOOGLE_API_KEY_RESOURCE_KEY,
+        GOOGLE_API_KEY_RESOURCE_TYPE, packageName)).thenReturn(0);
+    Context mockContext = Mockito.mock(Context.class);
+    Mockito.when(mockContext.getResources()).thenReturn(mockResources);
+    Mockito.when(mockContext.getPackageName()).thenReturn(packageName);
+    Assert.assertEquals(PROD_PWS_URL + "\0" + PROD_PWS_VERSION + "\0",
+        Utils.getDefaultPwsEndpointPreferenceString(mockContext));
+
+    Mockito.when(mockResources.getIdentifier(GOOGLE_API_KEY_RESOURCE_KEY,
+        GOOGLE_API_KEY_RESOURCE_TYPE, packageName)).thenReturn(resId);
+    Mockito.when(mockContext.getString(resId)).thenReturn(apiKey);
+    Assert.assertEquals(GOOGLE_PWS_URL + "\0" + GOOGLE_PWS_VERSION + "\0" + apiKey,
+        Utils.getDefaultPwsEndpointPreferenceString(mockContext));
+  }
+
+  @Test
+  public void testSetOptInPreference() throws Exception {
+
+  }
+
+  @Test
+  public void testSetPwsEndpointPreference() throws Exception {
+
+  }
+
+  @Test
+  public void testSetSharedPreferencesDefaultValues() throws Exception {
+
+  }
+
+  @Test
+  public void testSetPwsEndpointPwsClient() throws Exception {
+  }
+
+  @Test
+  public void testSetPwsEndpointPhysicalWebCollection() throws Exception {
```

Comment:
There is a Utils method that calls the set endpoint on the physical web collection.